Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/NMDataServiceTest.cs

```diff
@@ -0,0 +1,196 @@
+
+            IResult<BatchSearchResponse> res = new Result<BatchSearchResponse>();
+            A.CallTo(() => _fakefileShareService.FssBatchSearchAsync("", "", CorrelationId)).Returns(res);
+
+            string expectedstatus = "RanToCompletion";
+
+            Task<List<ShowFilesResponseModel>> ListshowFilesResponseModels = _fakeNMDataService.GetWeeklyBatchFiles(year, week, CorrelationId);
+
+            Assert.AreEqual(expectedstatus, ListshowFilesResponseModels.Status.ToString());
+
+        }
+
+        [Test]
+        public void WhenGetNMBatchFilesIsCalled_ThenShouldExecuteCatch()
+        {
+            int year = 2022;
+            int week = 15;
+
+            A.CallTo(() => _fakeAuthFssTokenProvider.GenerateADAccessToken(A<string>.Ignored)).Throws(new Exception());
+
+            IResult<BatchSearchResponse> res = new Result<BatchSearchResponse>();
+            A.CallTo(() => _fakefileShareService.FssBatchSearchAsync("", "", CorrelationId)).Returns(res);
+
+            Task<List<ShowFilesResponseModel>> result = _fakeNMDataService.GetWeeklyBatchFiles(year, week, CorrelationId);
+
+            Assert.That(result.IsFaulted, Is.True);
+        }
+
+
+        [Test]
+        public void WhenGetAllWeeksofYearIsCalled_ThenForCurrentYearShouldReturnWeeksPassedTillNow()
+        {
+            DateTimeFormatInfo dateTimeFormatInfo = DateTimeFormatInfo.CurrentInfo;
+
+            Calendar calender = dateTimeFormatInfo.Calendar;
+
+            int year = DateTime.Now.Year;
+
+            int totalWeeks = calender.GetWeekOfYear(new DateTime(year, DateTime.Now.Month, DateTime.Now.Day), dateTimeFormatInfo.CalendarWeekRule, dateTimeFormatInfo.FirstDayOfWeek);
+
+            List<KeyValuePair<string, string>> result = _fakeNMDataService.GetAllWeeksofYear(year, CorrelationId);
+
+            Assert.AreEqual(totalWeeks + 1, result.Count);
+        }
+        [Test]
+        public void WhenGetAllWeeksofYearIsCalled_ThenForPastYearShouldReturnAllWeeksThatYear()
+        {
+            DateTimeFormatInfo dateTimeFormatInfo = DateTimeFormatInfo.CurrentInfo;
+
+            Calendar calender = dateTimeFormatInfo.Calendar;
+
+            int year = DateTime.Now.Year - 1;
+
+            DateTime lastdate = new(year, 12, 31);
+
+            int totalWeeks = calender.GetWeekOfYear(lastdate, dateTimeFormatInfo.CalendarWeekRule, dateTimeFormatInfo.FirstDayOfWeek);
+
+            List<KeyValuePair<string, string>> result = _fakeNMDataService.GetAllWeeksofYear(year, CorrelationId);
+
+            Assert.AreEqual(totalWeeks + 1, result.Count);
+        }
+
+        [Test]
+        public void WhenGetPastYearsIsCalled_ThenShouldReturn4Records()
+        {
+            int yearsCount = 4;
+            List<KeyValuePair<string, string>> result = _fakeNMDataService.GetAllYears(CorrelationId);
+            Assert.AreEqual(yearsCount, result.Count);
+        }
+
+        [Test]
+        public void WhenGetPastYearsIsCalled_ThenShouldCheckMinYear()
```

Comment:
Should this be called WhenGetAllYearsIsCalled_......

Reply:
Test Name updated as WhenGetAllYearsIsCalled_ThenShouldCheckMinYear